### PR TITLE
NEO-1075: LeffNav->TopLink: Active state is now driven by currentUrl value

### PR DESCRIPTION
**Pull request by enrique-prado**
**Before tagging the team for a review, I have done the following:**
- [x ] reviewed my code changes
- [x] ensured that all tests pass
- [x ] added any important information to this PR (description, images, GIFs, ect.)
- [x] tagged Matt if any visual changes have occurred
- [x ] done an accessibility check on my work

We now update the TopLink active state if the currentUrl matches the url for that TopLinkItem

### CODE REVIEW

**Comment by matthewAvaya** [comment]
From your Gif example, it works as expected.

**Review by matthewAvaya — APPROVED** [review]

File: src/components/LeftNavigation/TopLinkItem/TopLinkItem.tsx [hunk 1]
```diff
@@ -18,14 +23,18 @@ export interface TopLinkItemProps {
...
   icon,
   id = useId(),
   disabled,
 }: TopLinkItemProps) => {
   const ctx = useContext(NavigationContext);
+  const [isActive, setIsActive] = useState(false);
+
+  useEffect(() => {
+    setIsActive(href === ctx.currentUrl);
+  }, [ctx.currentUrl]);
```
Comment (joe-s-avaya) [inline src/components/LeftNavigation/TopLinkItem/TopLinkItem.tsx:37]: I don't think that this is doing what you think it's doing. The URL in the stories is not updating, but the active state is, and this looks like it's comparing a partial URI to a full URI.

**Review by joe-s-avaya — CHANGES_REQUESTED** [review]
URL is not updated, which does not match what you describe in the PR description.

File: src/components/LeftNavigation/TopLinkItem/TopLinkItem.tsx (hunk 1 above)
Comment (yangAtSpoken) [reply src/components/LeftNavigation/TopLinkItem/TopLinkItem.tsx:37]: nit: put setIsActive and href in []

**Review by yangAtSpoken — APPROVED** [review]

File: src/components/LeftNavigation/TopLinkItem/TopLinkItem.tsx (hunk 1 above)
Comment (enrique-prado) [reply src/components/LeftNavigation/TopLinkItem/TopLinkItem.tsx:37]: added setIsActive and href to dep array.

**Review by joe-s-avaya — APPROVED** [review]

**Pull request merged**